compiler/src/dotty/tools/dotc/transform/SuperAccessors.scala

```diff
@@ -174,26 +174,20 @@ class SuperAccessors(thisPhase: DenotTransformer) {
     val sel @ Select(qual, name) = tree: @unchecked
     val sym = sel.symbol
 
-    /** If an accesses to protected member of a class comes from a trait,
-      *  or would need a protected accessor placed in a trait, we cannot
-      *  perform the access to the protected member directly since jvm access
-      *  restrictions require the call site to be in an actual subclass and
-      *  traits don't count as subclasses in this respect. In this case
-      *  we generate a super accessor instead. See SI-2296.
-      */
     def needsSuperAccessor =
-      ProtectedAccessors.needsAccessorIfNotInSubclass(sym) &&
+      ProtectedAccessors.needsAccessor(sym) &&
       AccessProxies.hostForAccessorOf(sym).is(Trait)
     qual match {
       case _: This if needsSuperAccessor =>
-        /*
-          * A trait which extends a class and accesses a protected member
-          *  of that class cannot implement the necessary accessor method
-          *  because jvm access restrictions require the call site to be in
-          *  an actual subclass and traits don't count as subclasses in this
-          *  respect. We generate a super accessor itself, which will be fixed
-          *  by the implementing class.  See SI-2296.
-          */
```

Comment:
I wonder why the reasoning in this comment can be ignored. Why do we not need the call-site to be in an actual subclass? Is it because protected members become public?

Then also have a test of Java protected members which are not made public by the JVM.

Reply:
> Why do we not need the call-site to be in an actual subclass? Is it because protected members become public?

Correct.  We can simply cast to the trait and call the method as a regular (non-super) call.

I'll add a Java protected member case.